src/main.rs

```diff
@@ -147,35 +230,126 @@ fn main() {
                     MouseMode::CameraLook => renderer.camera.move_on_orbit(mouse_delta),
                     MouseMode::CameraPan => renderer.camera.move_focus(mouse_delta),
                 }
+                ui_state.camera_target[0] = renderer.camera.target[0];
+                ui_state.camera_target[1] = renderer.camera.target[1];
+                ui_state.camera_target[2] = renderer.camera.target[2];
+                ui_state.camera_azimuth = renderer.camera.azimuth;
+                ui_state.camera_elevation = renderer.camera.elevation;
                 prev_mouse = position;
             }
             Event::MainEventsCleared => {
                 window.request_redraw();
             }
             Event::RedrawRequested(_) => {
                 // Redraw the application.
+                let frame_time_delta = prev_frame_time.elapsed();
+                prev_frame_time = imgui_context.io_mut().update_delta_time(prev_frame_time);
+
                 let frame = renderer
                     .swap_chain
                     .get_next_texture()
                     .expect("Timeout when acquiring next swap chain texture");
                 let mut commands = renderer
                     .device
                     .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
-                renderer.render(&mut commands, &frame.view, &rendering::defaults::axes(), &rendering::defaults::render_all_vertices(&rendering::defaults::axes()), true);
-                //renderer.render(&mut commands, &frame.view, &vertices, &indecies, false);
+                renderer.render(
+                    &mut commands,
+                    &frame.view,
+                    &rendering::defaults::axes(),
```

Comment:
is this actually a function call here? ::axes?  I haven't looked at the rest of the code.
Is this creating a new axes each frame or returning a static reference?